src/main/java/org/opentripplanner/standalone/config/framework/json/ParameterBuilder.java

```diff
@@ -197,27 +198,37 @@ public <T> List<T> asObjects(List<T> defaultValues, Function<NodeAdapter, T> map
     return buildAndListComplexArrayElements(defaultValues, mapper);
   }
 
+  /**
+   * @deprecated Avoid using required enum types, when adding/removing new Enum values
+   *             you potentially break forward/backward compatibility. If the new enum
+   *             value is used in the config, an earlier version of OTP can not read
+   *             the required value.
+   */
+  @Deprecated
   public <T extends Enum<T>> T asEnum(Class<T> enumType) {
     info.withRequired().withEnum(enumType);
-    return parseEnum(build().asText(), enumType);
+    // throws exception if enum value is missing
+    return parseRequiredEnum(build().asText(), enumType);
   }
 
   /** Get optional enum value. Parser is not case sensitive. */
   @SuppressWarnings("unchecked")
   public <T extends Enum<T>> T asEnum(T defaultValue) {
     info.withEnum((Class<? extends Enum<?>>) defaultValue.getClass());
     setInfoOptional(defaultValue);
-    // Do not inline the node, calling the build is required.
-    var node = build();
-    return exist() ? parseEnum(node.asText(), (Class<T>) defaultValue.getClass()) : defaultValue;
+    return parseOptionalEnum(build().asText(), (Class<T>) defaultValue.getClass())
+      .orElse(defaultValue);
   }
 
   public <T extends Enum<T>> Set<T> asEnumSet(Class<T> enumClass) {
     info.withOptional().withEnumSet(enumClass);
     List<T> result = buildAndListSimpleArrayElements(
       List.of(),
-      it -> parseEnum(it.asText(), enumClass)
-    );
+      it -> parseOptionalEnum(it.asText(), enumClass).orElse(null)
+    )
+      .stream()
+      .filter(Objects::nonNull)
```

Comment:
```suggestion
      it -> parseOptionalEnum(it.asText(), enumClass)
    )
      .stream()
      .filter(Optional::isPresent)
```

Reply:
If you're already using Optionals you can use their expressive power.

Reply:
Note! The method call that ends on line 228, dropping `.orElse(null)` is changing the type - does not work.

Reply:
I agree the cod does not look good, I will look at it
.